src/lib/parsers/sol-json.c

```diff
@@ -546,3 +546,20 @@ sol_json_double_to_str(const double value, char *buf, size_t len)
 
     return 0;
 }
+
+SOL_API bool
+sol_json_validate(struct sol_json_scanner *scanner, enum sol_json_type type)
```

Comment:
I'll add this comment: Check if scanner is pointing to a valid string with a json element of informed type. May or may not be NULL terminated